### PR TITLE
(BOLT-656) Add single-threaded executor for bolt-server

### PR DESCRIPTION
**Pull request by MikaelSmith**

### CODE REVIEW

File: lib/bolt/executor.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def initialize(concurrency = 1,
 
       @noop = noop
       @run_as = nil
-      @pool = Concurrent::ThreadPoolExecutor.new(max_threads: concurrency)
+      @pool = concurrency > 0 ? Concurrent::ThreadPoolExecutor.new(max_threads: concurrency) : nil
```
Comment (nicklewis) [inline lib/bolt/executor.rb]: I think if no concurrency is configured, we should use `Concurrent.global_immediate_executor`, which works like a normal Concurrent::Executor, but executes in the current thread.

Alternately, we could support passing *either* a Concurrent::Executor *or* a concurrency level here and let the caller decide. That would be useful if we wanted multiple Bolt::Executor instances backed by a single Concurrent::Executor

Comment (MikaelSmith) [reply lib/bolt/executor.rb]: I also wanted to avoid the promises, although maybe that's not much overhead.

Comment (MikaelSmith) [reply lib/bolt/executor.rb]: Alright, switched to using an ImmediateExecutor. I didn't see any negative impact in memory use from that (when running 1000 tasks).

File: lib/bolt_ext/server.rb [hunk 2]
```diff
@@ -10,10 +10,24 @@ class TransportAPI < Sinatra::Base
...
+  def initialize(app = nil)
+    @executor = Bolt::Executor.new(0, load_config: false)
+    super(app)
+  end
+
   get '/' do
     200
   end
 
+  get '/admin/gc' do
```
Comment (nicklewis) [inline lib/bolt_ext/server.rb]: Exposing this via the API concerns me... Could we gate this behind `RACK_ENV` for dev purposes only? I'm not quite as worried about the `/gc_stats` endpoint.

Comment (MikaelSmith) [reply lib/bolt_ext/server.rb]: Yeah, I can add that.

**Pull request merged**